uvtools/plot.py

```diff
@@ -471,6 +472,428 @@ def omni_view_gif(filenames, name='omni_movie.gif'):
+):
+    """
+    Plot a 2x2 grid of waterfalls showing all possible Fourier transforms.
+
+    Moving clockwise from the top-left, the plots are as follows:
+        time vs frequency
+        fringe-rate vs frequency
+        fringe-rate vs delay
+        time vs delay
+
+    Time units used for plotting are either JD or LST hours.
+    Frequency units used for plotting are MHz.
+    Fringe-rate units used for plotting are mHz.
+    Delay units used for plotting are ns.
+    
+    Note that these units may not correspond exactly to the units of the related
+    parameters passed. The parameter units for ``freqs`` and ``times`` or ``lsts``
+    are chosen to match the units typically used for recording metadata in
+    :class:`pyuvdata.UVData` objects.
+
+    Parameters
+    ----------
+    data: array-like of complex, or :class:`pyuvdata.UVData` instance
+        Object containing visibility data. If an array is passed, then ``freqs``
+        and either ``times`` or ``lsts`` must be provided, and the array must
+        have shape (``lsts.size``, ``freqs.size``). Otherwise, an ``antpairpol``
+        key must be provided.
+    antpairpol: tuple
+        (ant1, ant2, pol) tuple specifying the baseline and polarization to
+        pull data for if ``data`` is a :class:`pyuvdata.UVData` instance. Ignored
+        if ``data`` is an array-like object.
+    freqs: array-like of float
+        Frequencies corresponding to the observed data, in Hz. Required if ``data``
+        is an array-like object; ignored otherwise.
+    times: array-like of float
+        Observation times, in JD. Required if ``data`` is an array-like object and
+        ``lsts`` is not provided.
+    lsts: array-like of float
+        Observed LSTs, in radians. Required if ``data`` is an array-like object
+        and ``times`` is not provided.
+    time_or_lst: str, optional
+        Either "time" or "lst". Used to specify whether the time axis should be
+        in JD or LST. If ``data`` is an array-like object and only one of ``times``
+        or ``lsts`` is provided, then this parameter is ignored.
+    mode: str, optional
+        Plotting mode to use; must be a mode accepted by :func:`data_mode`. Default
+        is "log", which plots the base-10 logarithm of the absolute value of the data.
+    figsize: tuple of float, optional
+        Size of the figure to be produced, in inches. Default is 14x10.
+    dpi: float, optional
+        Dots-per-inch of the figure. Default is 100.
+    aspect: float or str, optional
+        Aspect ratio to use for each subplot. Default is "auto".
+    fontsize: float, optional
+        Font size to use for plotting labels, in points.
+    cmap: str or :class:`plt.cm.colors.Colormap` instance
+        Color map to be used when drawing waterfalls. Default is to have the choice
+        be based on the data mode selected: if ``mode`` is "phs", then "twilight"
+        is used; otherwise, "inferno" is used.
+    dynamic_range: dict, optional
```

Comment:
Is there a way to set the max or min? Need one or the other to fully set the color scale.

Reply:
It's currently handled under the hood. I can add extra options for controlling that at the function call, though it may have unexpected side effects if users give `vmin`/`vmax` parameters that are incompatible with the `mode` being used.

Reply:
you need two parameters to set the color scale, either max and min, or max and drange.  Here you've already got drange so just add a max.  If None, let the default float to max.

Reply:
So I've added `vmin` and `vmax` as optional parameters, and the color scale normalization is calculated using whichever two of (`vmin`, `vmax`, `dynamic_range`) are specified. If all three are provided, then it just uses `vmin` and `vmax`.

Reply:
I should make it clear that I added this functionality for `labeled_waterfall`, but not for `fourier_transform_waterfalls`, since the latter would require extra handling (it doesn't make sense to apply the same `vmin` or `vmax` parameter to two sets of data if one of them has been Fourier transformed). If `dynamic_range` is the only parameter provided, then it calculates the max under the hood and sets the minimum value according to that and the dynamic range.

Reply:
Ok. Lets roll with this and people can propose changes as they play.